deployments/operator/crd/bases/deviceplugin.intel.com_gpudeviceplugins.yaml

```diff
@@ -72,6 +72,13 @@ spec:
                 description: NodeSelector provides a simple way to constrain device
                   plugin pods to nodes with particular labels.
                 type: object
+              preferredAllocationPolicy:
+                description: 'PreferredAllocationPolicy sets the mode of allocating
```

Comment:
the formatting looks broken?

Reply:
This code is generated automatically by # make generate. So I am not sure whether need to modify manually? Could you  give a suggestion?

Reply:
I'd suggest make the comment below something like:

"PreferredAllocationPolicy sets the mode of allocating GPU devices on a node. See documentation
 for detailed description of the policies. Only valid when SharedDevNum is set."

And then about:
> Only valid when SharedDevNum is set.

We should add a validating webhook check about this in https://github.com/intel/intel-device-plugins-for-kubernetes/blob/48dcc8eb6d3481840160f3b9168701c6edf1a474/pkg/apis/deviceplugin/v1/gpudeviceplugin_webhook.go#L88-L96

What do you think?

Reply:
Yeah, the comment you suggest is better. But do we need to modify 

> Only valid when SharedDevNum is set.
 to
> Only valid when SharedDevNum > 1 is set.

Because the default value of SharedDevNum is 1, set it to 1 is the same as not set.

Reply:
> Because the default value of SharedDevNum is 1,

Right, I forgot the default value. I'm fine with "> 1 is set".

Reply:
I have an additional

Reply:
I have a supplement here, if we 

> add a validating webhook check about this 

Then, we also need to add a check in 
`func main() {
...
        if opts.sharedDevNum < 1 {
                klog.Error("The number of containers sharing the same GPU must greater than zero")
                os.Exit(1)
        }

        if opts.sharedDevNum == 1 && opts.resourceManagement {
                klog.Error("Trying to use fractional resources with shared-dev-num 1 is pointless")
                os.Exit(1)
        }

        klog.V(1).Infof("GPU device plugin started with %s preferred allocation policy", opts.preferredAllocationPolicy)
...
`

Reply:
The validating webhook check is added.